services/sharedchannel/sync_send.go

```diff
@@ -133,70 +197,99 @@ func (scs *Service) processTask(task syncTask) error {
 		remotes = []*model.RemoteCluster{rc}
 	}
 
-	errs := merror.New()
 	for _, rc := range remotes {
-		if err := scs.updateForRemote(task.channelId, rc); err != nil {
-			errs.Append(err)
+		rtask := task
```

Comment:
Why the copy?

Reply:
I am modifying the source task within a loop and passing the result to a method within that loop.  While technically I could modify the original and let pass-by-value do the copy I think that makes the code more fragile. If for some reason someone makes task a pointer it will break in a subtle way.

Reply:
I am modifying the original task within a loop and using the copy in a method.  Minor style thing in that changing the incoming task contents feels wrong from a code fragility perspective. I guess it doesn't matter.

Reply:
I guess the question is: should the retry counter increment per remote?

Reply:
Yes, it must since it fails per remote.